packages/puppeteer-core/src/api/Page.ts

```diff
@@ -437,7 +437,7 @@ export interface PageEvents extends Record<EventType, unknown> {
   [PageEvent.Response]: HTTPResponse;
   [PageEvent.RequestFailed]: HTTPRequest;
   [PageEvent.RequestFinished]: HTTPRequest;
-  [PageEvent.RequestServedFromCache]: HTTPRequest;
+  [PageEvent.RequestServedFromCache]: HTTPRequest | undefined;
```

Comment:
Revert this, this should never happen.

Reply:
This can occur. See https://crbug.com/750469. It shouldn't be resolved.

Reply:
The user can't do anything if we return `undefined`, they can't understand why or from where it's coming without looking at CDP logs, so changing to only emit when we already have the request in internal map is the correct way of doing it. Also braking API change.

Reply:
`RequestServedFromCache` _needs_ to be emitted whenever that event comes in. Whether it exists or not in the internal map is a bug due to `https://crbug.com/750469`.

This isn't a breaking change since this is what happens at runtime.

Reply:
That not true we want to emit the event whenever we know that are `HTTPRequest` comes from cache. 
What you are proposing gives no user value but they will need to add extra checks for this.

Reply:
The cache is from the internal, Chromium cache. Not the one in Puppeteer.

Reply:
Again I think you are missing the point - As a user this event will provide no value to me as it will be empty and can't do anything with it - log info or intercept it, thus I will always need to ignore it. We should ignore internally and save the user a check, also we should keep the API the same as it is now and fix our implementation to match it, whenever possible.